eng/common/pipelines/templates/steps/check-spelling.yml

```diff
@@ -0,0 +1,27 @@
+# Checks spelling of files that changed between the current state of the repo
+# and some ref (branch, tag, etc.) or commit hash. Only runs on PRs.
+# ContinueOnError   - true: Pipeline warns on spelling error 
+#                     false: Pipeline fails on spelling error
+# TargetBranch      - Target ref (e.g. master) to compare to create file change
+#                     list.
+# CspellConfigPath  - Path to cspell.json config location
+
+parameters:
+  ContinueOnError: true
+  TargetBranch: $(System.PullRequest.TargetBranch)
+  SourceBranch: HEAD
+  CspellConfigPath: ./.vscode/cspell.json
+
+steps:
+  - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+    - task: PowerShell@2
+      displayName: Check spelling (cspell)
+      continueOnError: ${{ parameters.ContinueOnError }}
+      inputs: 
+        targetType: filePath
+        filePath: eng/common/scripts/check-spelling-in-changed-files.ps1
```

Comment:
nit: [Verb-Noun.ps1] is adiomatic for PowerShell entrypoint scripts.

Reply:
@weshaggard initially proposed changing it. I had it called `Test-Spelling.ps1` initially... I hold very lightly the opinion that pouring our scripts into the verb model makes things easier for all end users.

Reply:
https://docs.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands?view=powershell-7.1 that is the guidance for the verb-noun model but I think makes some sense for cmdlets but  I personally don't think it makes sense for functions or script names. Trying to fit that model is pretty difficult and I think it usually ends up with less understandability for our usages than just naming a script or a function like you would want to call it. So I don't recommend using that guideline for most of our scenarios.